examples/ProjPlaneCurve.jl

```diff
@@ -0,0 +1,534 @@
+  degree::Int64                           # degree of the equation of the curve
+  dimension::Int64                        # dimension of the curve (as a variety)
+  components::Dict
+  function ProjPlaneCurve(eq::Oscar.MPolyElem_dec{S}) where {S <: FieldElem}
+    r = new{S}()
+    r.eq = eq
+    if nvars(parent(r.eq)) != 3
+       error("The defining equation must belong to a ring with three variables")
+    elseif isconstant(r.eq) == true
+       error("The defining equation must be non constant")
+    else
+       if ishomogenous(r.eq) == false
+          error("The defining equation is not homogeneous")
+       end
+    end
+    r.degree = -1                      # -1 when it is not computed yet
+    r.dimension = 1                    # since C is a plane curve, the dimension is always 1
+    r.components = Dict{ProjPlaneCurve{S}, Int64}()
+    return r
+  end
+  function ProjPlaneCurve(eq::Oscar.MPolyElem{S}) where {S <: FieldElem}
+     R = grade(parent(eq))
+     return ProjPlaneCurve(R(eq))
+  end
+end
+
+function Base.show(io::IO, C::ProjPlaneCurve)
+  if !get(io, :compact, false)
+     println(io, "Projective plane curve defined by ", C.eq)
+  else
+     println(io, C.eq)
+  end
+end
+
+################################################################################
+# hash function
+
+function Base.hash(C::ProjPlaneCurve, h::UInt)
+  F = 1//lc(C.eq)*C.eq
+  return hash(F, h)
+end
+
+################################################################################
+
+################################################################################
+# Compute the degree of the equation of the curve if not already known,
+# and show it.
+
+@doc Markdown.doc"""
+degree(C::ProjPlaneCurve)
+> Returns the degree of the defining polynomial of C.
+"""
+function Oscar.degree(C::ProjPlaneCurve)
+  if C.degree >= 0
+     return C.degree
+  else
+     C.degree = total_degree(C.eq.f)
+     return C.degree
+  end
+end
+
+################################################################################
+# Compute the Jacobian Ideal of C
+
+@doc Markdown.doc"""
+jacobi_ideal(C::ProjPlaneCurve)
+> Returns the Jacobian ideal of the defining polynomial of C.
+"""
+function Oscar.jacobi_ideal(C::ProjPlaneCurve)
+ return jacobi_ideal(C.eq)
+end
+
+################################################################################
+# To check if a point is smooth: return true if the point is a smooth point on
+# the curve C, and false  if it is singular, and an error if it is not on the
+# curve.
+
+@doc Markdown.doc"""
+issmooth(C::ProjPlaneCurve{S}, P::Oscar.Geometry.ProjSpcElem{S}) where S <: FieldElem
+> Returns an error if P is not a point of C, false if P is a singular point of C, and true if P is a smooth point of C.
+"""
+function Oscar.issmooth(C::ProjPlaneCurve{S}, P::Oscar.Geometry.ProjSpcElem{S}) where S <: FieldElem
+  if dim(P.parent) != 2
+     error("The point needs to be in a projective two dimensional space")
+  elseif evaluate(C.eq, P.v) != 0
```

Comment:
Doesn't the evaluation really give a field element? Then it should be:
```suggestion
  elseif !iszero(evaluate(C.eq, P.v))
```